### PR TITLE
start eic_report fn for #5

### PR DESCRIPTION
**Pull request by mpadge**

### CODE REVIEW

File: R/eic-report.R [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    return (q)
+}
+
+#' Apply the query above to extract data from GitHub GraphQL API, and
+#' post-process into a `data.frame`.
+#' @noRd
+devguide_eic_gh_data <- function () {
+
+    token <- get_gh_token ()
+    gh_cli <- ghql::GraphqlClient$new (
```
Comment (maelle) [inline R/eic-report.R:73]: why not use `gh::gh_gql()`? (just curious)

Comment (mpadge) [reply R/eic-report.R]: No real reason; i just copy-pasta'd code from elsewhere to get started. I've now changed, and was suprised at the effort it took. I think `ghql` does a much better job of hiding the ugliness of nested list nightmares, simply through naming everything. `gh` is admittedly more directly representative of a query, but it then leaves crucial parts unnamed, and thus only accessible through loads of `apply`-type calls that `ghql` does not require. I never really appreciated that difference before.

And the named/unnamed behaviour reflects this:
https://github.com/ropensci-org/devguider/blob/327eebb4dd33df2d856f33fed3b25bfdb9a95eff/R/eic-report.R#L29
which somehow translates in `ghql` to nothing at all, and so named list items simply jump to everything within there, whereas `gh` makes each `... on Issue` item a distinct *unnamed* list item. Which is annoying.

**Comment by mpadge** [comment]
@maelle Can you please try running `devguide_eic_report()`, and letting me know what you think? I think it's got most of the information there that we need. Thanks!

**Comment by maelle** [comment]
@mpadge awesome!

- First of all it's nice that it just worked!
- Interesting that some assignees are not editors, could the two columns be near each other and could "editor" also use the name to make it easier to notice discrepancies?
- What's the final output expected to be, this data.frame? Or a tibble? Or an actual report?
- I wonder whether it'd be easy to create a visualization of submissions over time, with lines from "created at" to "last edited" + reviews due date (not easy to notice if the reviews have been submitted as it'd add the Airtable querying :thinking: ).

**Comment by maelle** [comment]
a report might be nice as we could make a list of stuff to pay attention to, like submissions still at the prep stage.

**Comment by mpadge** [comment]
All good ideas; shall do! Thanks!

**Comment by maelle** [comment]
Please tell me if I can help :-)

**Comment by mpadge** [comment]
@maelle Can you please take this for a spin now? The whole thing is just:

``` r
dat <- devguider_eic_report ()
```

You can also re-generate the HTML report with the internal function, `open_gt_table(dat)`.

**Comment by maelle** [comment]
> In devguide_eic_report() :
  The following issue currently has multiple 'stage' labels:
   551
   
So useful!!

**Comment by mpadge** [comment]
Comment from @noamross:

> I think it would be useful for there to be some quantity or signal (color) that indicates what needs attention, such as a long time since the status has changed. The dates inform this a little but not at a glance

I agree, and propose that the current yellow columns to the left (core data on [issue] "number", "title", "label") be done in some kind of gradated scale proportional to time since status last changed, with scales for 3, 4, 5 done differently to 0, 1, 2, because the latter (0, 1, 2 = us organising stuff) should be quicker than the review phases (3,4, 5). Thoughts @maelle?

(And note i do not think we should over-complexify here by, for example, scaling to empirical data on how long each phase typically lasts - i'm confident we'll get clear visual indications here by keeping it as simple as possible.)

**Comment by maelle** [comment]
I agree! A checklist would still be nice I think, to be sure one has looked at all important aspects.

**Comment by mpadge** [comment]
@maelle Ready to approve now, please. The color scheme is ... loud ... but i'm not intending to worry too much about that. One important point of review for both you and @noamross are the following encoded assumptions of typical time scales each stage should take:
https://github.com/ropensci-org/devguider/blob/92e335a1790e955cdaf8e8bbbdf081bd28f2fe97/R/eic-report.R#L428-L436
These are translated into an "urgency" by dividing elapsed days by these numbers, maxxing all urgency values at 5, and colouring accordingly. So an urgency of 4 means something has taken 4 times longer than expected. Happy to tweak if either of you want.

**Comment by maelle** [comment]
Looks good! Should I open another issue reg a checklist to go with this?

**Comment by mpadge** [comment]
> Looks good! Should I open another issue reg a checklist to go with this?

Yes, please, and is it okay if you do that one?

**Pull request merged**